### PR TITLE
Add reduction `distinct_count` benchmark

### PR DESCRIPTION
**Pull request by ttnghia**
This adds a simple benchmark for reduction `distinct_count`.

### CODE REVIEW

File: cpp/benchmarks/reduction/distinct_count.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#include <nvbench/nvbench.cuh>
+
+template <typename Type>
+static void bench_reduction_distinct_count(nvbench::state& state, nvbench::type_list<Type>)
+{
+  cudf::rmm_pool_raii pool_raii;
+
+  auto const dtype          = cudf::type_to_id<Type>();
+  const auto size           = static_cast<cudf::size_type>(state.get_int64("num_rows"));
```
Comment (karthikeyann) [inline cpp/benchmarks/reduction/distinct_count.cpp:30]: prefer East const.

Comment (ttnghia) [reply cpp/benchmarks/reduction/distinct_count.cpp:30]: Copy and past error :smile:

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Review by elstehle — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**